test/code/plugins/oms_common_test.rb

```diff
@@ -209,6 +357,38 @@ def test_get_hostname_in_containerized_agent
       assert_equal(test_hostname, hostname, 'get_hostname should read from Socket.gethostname')
     end
 
+    def test_get_hostname_mocked
+      $log = MockLog.new
+      
+      flexmock(Socket, :strict, Socket => :gethostname, :gethostname => "myhostname") do
+        hostname = Socket.gethostname
+        assert_equal("myhostname", hostname, "Could not get the hostname")
+        assert(hostname.size > 0, "Hostname returned is empty")
+        load "#{ENV['BASE_DIR']}/source/code/plugins/oms_common.rb"
+        hostname = Common.get_hostname # TODO add flexmock thing to have Socket.gethostname return an IP (or better yet, return the results of Socket.ip_address_list
```

Comment:
Remove TODO commments that you have taken care of

Reply:
This would also be a good place to assert that nothing gets raised, since "myhostname" is valid.